module/Finna/src/Finna/RecordDriver/Feature/ContainerFormatTrait.php

```diff
@@ -0,0 +1,269 @@
+<?php
+/**
+ * Common functionality for container record formats.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+namespace Finna\RecordDriver\Feature;
+
+use Finna\Record\Loader;
+use Finna\RecordDriver\PluginManager;
+use VuFind\RecordDriver\AbstractBase;
+
+/**
+ * Common functionality for container record formats.
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+trait ContainerFormatTrait
+{
+    /**
+     * Names of specific encapsulated record XML elements.
```

Comment:
Kaikki ei välttämättä ole XML:ää. Pitäisin mahdollisena myös sen, että olisi vaikka JSON-muotoisia tietueita (kuten MARC-tietueet ovat jatkossa). Näin ollen ainakin pitäisi tietuetta kaivelevat kohdat tehdä niin, että ne voi yliajaa ilman muuta kopiointia. Ehkä tätä taulukkomappausta selkeämpää olisi, että olisi getterit tyyliin `getEncapsulatedRecordFormat($record)`, ja oletustoteutus XML-tietueelle, jossa format-elementti.